src/core/service.c

```diff
@@ -1644,6 +1647,8 @@ static void service_enter_signal(Service *s, ServiceState state, ServiceResult f
 
 fail:
         log_unit_warning_errno(UNIT(s), r, "Failed to kill processes: %m");
+        if (s->stop_result == SERVICE_SUCCESS)
+                s->stop_result = SERVICE_FAILURE_RESOURCES;
```

Comment:
maybe add a helper service_update_stop_result() or so, which does this above two lines? i.e. sets stop_result() if it isn't set yet, but leaves it as is if it is already set? it appears to me this logic should be used at various places...

(And yeah, we should have a similar function for the service' main result too, and use that everywhere)